### PR TITLE
file-based portfolio synchronisation

### PR DESCRIPTION
**Pull request by MichaelRawson**
See #462. We currently synchronise child Vampires and their proof printing by UNIX semaphores, which are quite complex beasts. However, C11 (and therefore C++17) can synchronise on a file by passing an "x"clusive flag to `fopen`: if the file exists, `fopen` will fail. Note that the standard only says "to the extent the platform supports exclusive access", but it seems to work quite well in practice.

Therefore:
1. We determine a path that should be written to. This is either what the user requested or a temporary file.
2. We delete it if it exists!
3. The first child to *not* find that file creates it and writes its proof there.
4. Other children find the file and do not print a proof.
5. The parent Vampire knows the path and reads from it once all children have finished (one way or another).

I think this is simpler and certainly more portable, but of course it needs lots of testing: it would be a silly way to lose problems.

### CODE REVIEW

**Comment by MichaelRawson** [comment]
To explain the signal-handling change: currently we handle many different signals with very similar but slightly different code. This looks more like organic growth over time rather than something intentionally complex, so I've streamlined it. There are now only two broad classes of signal we handle:
1. Exit now, silently: `SIGHUP` (portfolio parent died), `SIGINT` (Ctrl-C), `SIGTERM` (polite termination), and `SIGXCPU` (we're being warned that we've used an external resource limit and are about to be killed). `SIGXCPU` has moved from the other class, I don't think we want to special-case this.
2. Crash of some kind: everything else, including `SIGQUIT` as apparently that's an abnormal halt intended to be used for debugging. Here we print a stack trace, indicate something to Spider, etc.

**Comment by quickbeam123** [comment]
Since you already polished the signals, could you please give me back the feature we used to have, that Vampire reports statistics (if asked) and termination reason on CTRL+C?

File: Lib/System.cpp [hunk 1]
```diff
@@ -115,13 +115,16 @@ void handleSignal (int sigNum)
     TERMINAL_SIGNAL_HANDLED = true;
     Shell::reportSpiderFail();
     if(Shell::outputAllowed(true)) {
-      std::cout << getpid() << " Aborted by signal " << signalToString(sigNum);
+      Shell::addCommentSignForSZS(std::cout);
+      std::cout << "Aborted by signal " << signalToString(sigNum);
```
Comment (MichaelRawson) [inline Lib/System.cpp:119]: Do we perhaps still want the PID to work out which process has received the signal? But no strong feelings, just wondered if this was an oversight.

Comment (MichaelRawson) [reply Lib/System.cpp:119]: Ahah, my mistake. Thanks!

**Pull request merged**